go-apps/meep-loc-serv/server/loc-serv.go

```diff
@@ -311,6 +361,287 @@ func registerUser(userAddress string, event []UserEventType, subsIdStr string) {
+
+				switch *distanceCheck.Subscription.Criteria {
+				case ALL_WITHIN_DISTANCE:
+					if float32(distance) < distanceCheck.Subscription.Distance {
+						returnAddr[monitoredAddr] = &distResp
+					} else {
+						skipThisSubscription = true
+						break
+					}
+				case ALL_BEYOND_DISTANCE:
+					if float32(distance) > distanceCheck.Subscription.Distance {
+						returnAddr[monitoredAddr] = &distResp
+					} else {
+						skipThisSubscription = true
+						break
+					}
+				case ANY_WITHIN_DISTANCE:
+					if float32(distance) < distanceCheck.Subscription.Distance {
+						returnAddr[monitoredAddr] = &distResp
+					}
+				case ANY_BEYOND_DISTANCE:
+					if float32(distance) > distanceCheck.Subscription.Distance {
+						returnAddr[monitoredAddr] = &distResp
+					}
+				default:
+				}
+			}
+			if skipThisSubscription {
+				continue
+			}
+			if len(returnAddr) > 0 {
+				subsIdStr := strconv.Itoa(subsId)
+
+				var distanceNotif SubscriptionNotification
+				distanceNotif.DistanceCriteria = distanceCheck.Subscription.Criteria
+				distanceNotif.IsFinalNotification = false
+				distanceNotif.Link = distanceCheck.Subscription.Link
+				var terminalLocationList []TerminalLocation
+				for terminalAddr, distanceInfo := range returnAddr {
+					var terminalLocation TerminalLocation
+					terminalLocation.Address = terminalAddr
+					var locationInfo LocationInfo
+					locationInfo.Latitude = nil
+					locationInfo.Latitude = append(locationInfo.Latitude, distanceInfo.Latitude)
+					locationInfo.Longitude = nil
+					locationInfo.Longitude = append(locationInfo.Longitude, distanceInfo.Longitude)
+					locationInfo.Shape = 2
+					seconds := time.Now().Unix()
+					var timestamp TimeStamp
+					timestamp.Seconds = int32(seconds)
+					locationInfo.Timestamp = &timestamp
+					terminalLocation.CurrentLocation = &locationInfo
+					retrievalStatus := RETRIEVED
+					terminalLocation.LocationRetrievalStatus = &retrievalStatus
+					terminalLocationList = append(terminalLocationList, terminalLocation)
+				}
+				distanceNotif.TerminalLocation = terminalLocationList
+				distanceNotif.CallbackData = distanceCheck.Subscription.ClientCorrelator
+				var inlineDistanceSubscriptionNotification InlineSubscriptionNotification
+				inlineDistanceSubscriptionNotification.SubscriptionNotification = &distanceNotif
+				sendSubscriptionNotification(distanceCheck.Subscription.CallbackReference.NotifyURL, inlineDistanceSubscriptionNotification)
+				log.Info("Distance Notification"+"("+subsIdStr+") For ", returnAddr)
+			}
+		}
+	}
+}
+
+func checkNotificationAreaCircle(addressToCheck string) {
+
+	//only check if there is at least one subscription
+	mutex.Lock()
+	defer mutex.Unlock()
+	//check all that applies
+	for subsId, areaCircleCheck := range areaCircleSubscriptionMap {
+		if areaCircleCheck != nil && areaCircleCheck.Subscription != nil {
+			//decrement the next time to send a message
+			areaCircleCheck.NextTts--
```

Comment:
Is Frequency defined in seconds?
Are you assuming that this function will be called once every second and using a counter instead of a ticker?

Reply:
comments above explain the reason why i did so... the same applies to other subscription driven by a frequency/period...

Reply:
The issue here is that this function is called from the SBI callback (updateUserInfo), and not on a 1 second ticker.